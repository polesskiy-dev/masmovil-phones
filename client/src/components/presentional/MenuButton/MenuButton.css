.menu-button__wrapper {
    padding: 0 32px;
    border-left: 1px solid var(--app-menu-text-color);
}

.menu-button {
    background: none;
    border: none;
    font: inherit;
    cursor: pointer;
    outline: inherit;

    color: var(--app-menu-text-color);
}

.menu-button:hover{
    text-decoration: underline;
}
