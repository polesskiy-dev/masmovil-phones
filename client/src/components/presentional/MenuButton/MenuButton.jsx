import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import './MenuButton.css';

const MenuButton = ({ caption, messageId, onClick }) => (
  <div className="menu-button__wrapper">
    <button type="button" className="menu-button" onClick={() => onClick()}>
      <FormattedMessage id={messageId} /> {caption}
    </button>
  </div>
);

MenuButton.propTypes = {
  caption: PropTypes.string.isRequired,
  messageId: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default MenuButton;
