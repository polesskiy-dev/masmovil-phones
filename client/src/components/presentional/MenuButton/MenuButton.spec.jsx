import React from 'react';
import renderer from 'react-test-renderer';
import _ from 'lodash/fp';

import withDefaultIntl from '../../../helpers/with-default-intl';
import MenuButton from './MenuButton';

it('<PhoneDetailComponent/ > renders Theme button correctly with all required fields', () => {
  const tree = renderer
    .create(
      withDefaultIntl(
        <MenuButton
          caption=""
          onClick={_.noop}
          messageId="AppMenuHeader.theme"
        />
      )
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
