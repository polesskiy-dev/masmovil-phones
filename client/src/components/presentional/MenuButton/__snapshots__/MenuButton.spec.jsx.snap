// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhoneDetailComponent/ > renders Theme button correctly with all required fields 1`] = `
<div
  className="menu-button__wrapper"
>
  <button
    className="menu-button"
    onClick={[Function]}
    type="button"
  >
    <span>
      Theme:
    </span>
     
    
  </button>
</div>
`;
